### PR TITLE
Address Space ID Upgrade for CAAS

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch augments the upgrade step for updating addresses to use space ID instead of name and provider ID.

The addresses in collections _cloudservices_ and _cloudcontainers_ are now updated. These will not have space names as CAAS is currently space-less, but they have the "0" space ID added.

## QA steps

- Bootstrap a CAAS model and deploy any unit(s)/bundle.
- Update the operator image using the make job from this patch.
- Run `juju upgrade-controller` and observe success without error.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Upgrade Steps for CAAS Address Space IDs](https://trello.com/c/f8BRUOyY/1887-upgrade-steps-for-caas-address-space-ids)

File: state/upgrades.go [hunk 1]
```diff
@@ -2595,64 +2595,158 @@ func EnsureRelationApplicationSettings(pool *StatePool) error {
...
+	if err = col.Find(nil).All(&docs); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var ops []txn.Op
+	for _, doc := range docs {
+		for i := range doc.Addresses {
+			// CAAS addresses at this point in time are space-less.
+			// We just need to ensure that they all have the zero ID.
+			doc.Addresses[i].SpaceID = "0"
```
Comment (achilleasa) [inline state/upgrades.go]: Not `network.DefaultSpaceID`?

File: state/upgrades.go [hunk 2]
```diff
@@ -2595,64 +2595,158 @@ func EnsureRelationApplicationSettings(pool *StatePool) error {
...
+	for _, doc := range docs {
+		if doc.Address == nil {
+			continue
 		}
-		return errors.Trace(st.db().RunTransaction(ops))
-	}))
+
+		// CAAS addresses at this point in time are space-less.
+		// We just need to ensure that they all have the zero ID.
+		doc.Address.SpaceID = "0"
```
Comment (achilleasa) [inline state/upgrades.go]: Same comment here about using constants

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**